src/read.c

```diff
@@ -2013,6 +2030,108 @@ static avifBool avifParseToneMappedImageBox(avifGainMapMetadata * metadata, cons
 }
 #endif // AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP
 
+#if defined(AVIF_ENABLE_EXPERIMENTAL_SAMPLE_TRANSFORM)
+// bit_depth is assumed to be 2 (32-bit).
+static avifResult avifParseSampleTransformTokens(avifROStream * s, avifSampleTransformExpression * expression)
+{
+    uint8_t tokenCount;
+    AVIF_CHECK(avifROStreamRead(s, &tokenCount, /*size=*/1)); // unsigned int(8) token_count;
+    AVIF_CHECKERR(avifArrayCreate(expression, sizeof(expression->tokens[0]), tokenCount), AVIF_RESULT_OUT_OF_MEMORY);
+
+    for (uint32_t t = 0; t < tokenCount; ++t) {
+        avifSampleTransformToken * token = (avifSampleTransformToken *)avifArrayPush(expression);
+        AVIF_ASSERT_OR_RETURN(token != NULL);
```

Comment:
`AVIF_ASSERT_OR_RETURN` is not appropriate here. This should return `AVIF_RESULT_OUT_OF_MEMORY`.

Reply:
I disagree, `avifArrayCreate()` was just called with sufficient `initialCapacity`. But your suggestion is easier to maintain I guess.

Reply:
You are right. I missed that point. Sorry. This should be an assertion.

Note: This array can be a fixed-sized dynamically-allocated array. It's not necessary to use avifArray.